odd_collector/domain/plugin.py

```diff
@@ -333,6 +333,18 @@ class CKANPlugin(WithHost, WithPort):
     token: Optional[SecretStr] = None
 
 
+class OpensearchPlugin(WithHost):
+    type: Literal["opensearch"]
+    host: str
+    port: int
+    http_compress: bool = True
+    use_ssl: bool = True
+    username: str
```

Comment:
`username: str` => `username: Optional[str]`
`password: SecretStr` => `password: Optional[SecretStr]`

One more thing to notice, here you can set default value for cases  when use omit that fileds, by default it is `None`, but you can set something that you opensearch client receives by defaulf.